packaging/nix/package.nix

```diff
@@ -0,0 +1,95 @@
+{
+  # Dependencies:
+  lib,
+  buildFHSEnv,
+  umu-launcher-unwrapped,
+
+  # Public API:
+  version ? null,
+  withTruststore ? args.truststore or true,
+  withDeltaUpdates ? true,
+
+  # We will manually handle renamed/deprecated args below
+  ...
+}@args:
+let
+  unknownArgs = builtins.removeAttrs args [
+    "lib"
+    "buildFHSEnv"
+    "umu-launcher-unwrapped"
+    "version"
+    "withTruststore"
+    "withDeltaUpdates"
+    "truststore"
+    "cbor2"
+  ];
+
+  # Safely print args & values, based on `lib.options.showDefs`
+  showArgs = lib.concatMapAttrsStringSep "\n" (
+    name: value:
+    let
+      # Attempt to pretty print the value
+      prettyEval = builtins.tryEval (
+        lib.generators.toPretty { } (
+          lib.generators.withRecursion {
+            depthLimit = 10;
+            throwOnDepthLimit = false;
+          } value
+        )
+      );
+      # Split it into its lines
+      lines = lib.splitString "\n" prettyEval.value;
+      lineCount = builtins.length lines;
+      # Only display the first 5 lines, and indent them for better visibility
+      prettyValue = lib.concatStringsSep "\n    " (
+        lib.take 5 lines ++ lib.optional (lineCount > 5) "..."
+      );
+      finalValue = lib.optionalString (prettyEval.success && lineCount > 0) (
+        if lineCount == 1 then ": ${prettyValue}" else ":\n    ${prettyValue}"
+      );
+    in
+    "- ${name}${finalValue}"
+  );
+
+  overridden = umu-launcher-unwrapped.overrideAttrs (old: {
+    version =
+    if version == null then
+      lib.warn (
+        "umu-launcher: you have not specified a `version`. "
+        + "Using ${lib.generators.toPretty { } old.version}."
+      ) old.version
+    else
+      version;
+
+    withTruststore =
+      lib.warnIf (args ? truststore)
+        "umu: the argument `truststore` has been renamed to `withTruststore`"
+        withTruststore;
+
+    withDeltaUpdates =
+      lib.warnIf (args ? cbor2)
+        "umu: the argument `cbor2` has never had any effect. The new argument `withDeltaUpdates` should be used instead"
+        withDeltaUpdates;
+  });
+in
+# Emulate not having `...` in args
+assert lib.assertMsg (unknownArgs == { }) ''
+  umu: called with unexpected arguments:
+  ${showArgs unknownArgs}
+'';
```

Comment:
We should just remove all of this imo

Reply:
Having this means removing the `...` later will not be a breaking change

Reply:
Boo hoo just make the breaking change